### PR TITLE
RFC for custom Response creation.

### PR DESCRIPTION
**Pull request by archolewa**
A design for support of custom response serialization.

### CODE REVIEW

File: rfc/resultSetSerializability.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    a. A `ResponseData` class that contains all of the data and helper methods
+        currently living on `Response`.
+    b. An implementation of `ResponseWriter` for each of
+        `Response::writeJsonResponse`, `Response::writeJsonApiResponse`,
+        `Response::writeCsvResponse`.
+
+3. Define an interface `ResponseWriterSelector` with a single method:
+    ```java
+        ResponseWriter select(DataApiRequest request);
+    ```
```
Comment (michael-mclawhorn) [inline rfc/resultSetSerializability.md]: Should this be optional?

**Comment by QubitPi** [comment]
Why not consider `TableApiRequest` and `DimensionApiRequest`, as well?

**Comment by archolewa** [comment]
@QubitPi I've updated the interface for `ResponseWriter` to take an `ApiRequest` rather than a `DataApiRequest`, because you're right, we should allow the ability to fully customize the other servlets.

However, fully generalizing this to work for the other endpoints will require more than just having the interface take an `ApiRequest`. For one thing, we'd have to generalize `ResponseData` somehow to accomodate for the very differently structured responses that the other servlets return. So, adding support for rewriting the other endpoints is outside the scope of this RFC.

**Comment by archolewa** [comment]
@kevinhinterlong Should probably go under docs.

**Comment by archolewa** [comment]
@kevinhinterlong I've moved the rfc folder under `docs`.

**Pull request closed without merging**